simplified-books-core/src/main/java/org/nypl/simplified/books/core/SimplifiedBearerToken.kt

```diff
@@ -0,0 +1,38 @@
+package org.nypl.simplified.books.core
+
+import org.json.JSONException
+import org.json.JSONObject
+import java.net.URI
+import java.net.URISyntaxException
+import java.util.Calendar
+import java.util.Date
+
+internal data class SimplifiedBearerToken(
+    val accessToken: String,
+    val expiration: Date,
+    val location: URI) {
+
+  companion object Factory {
+    fun withJSONObject(jsonObject: JSONObject): SimplifiedBearerToken? {
+      try {
+        if (jsonObject.getString("token_type") != "Bearer") {
```

Comment:
Should we add any kind of logging in the interest of any potential issues as a completely new process implemented in the app to potentially help the server team? Any warning-level issues are automatically uploaded to Bugsnag.
This was my yet-untested attempt:
```
import org.slf4j.LoggerFactory
...
private companion object {
    val LOG = LoggerFactory.getLogger(AnnotationsManager::class.java)!!
}
```

Reply:
I'll add logging for this one!

Reply:
@gioneill Actually, I already have logging for this one. If the token fails to parse, the caller of `withJSONObject` throws an exception with a message indicating as such and that is then logged after the exception is caught. I prefer that to having the logging in the parsing code itself because it's not really the parsing that's the problem—it's the data the caller is passing to it. As such, I think we're all set!